kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -170,6 +170,194 @@ static int findMaxColumnId(StructType schema) {
     return maxColumnId;
   }
 
+  /**
+   * The following validation is performed on the updated schema
+   *
+   * <ul>
+   *   <li>Every field has column IDs and physical names, and that these are unique
+   *   <li>Array/map types have valid nested field IDs defined if icebergCompatV2 is enabled
+   *   <li>Physical column names are preserved across the update
+   * </ul>
+   *
+   * @param currentColumnPathToField column path to field (both nested and leaf fields) for current schema
+   * @param newColumnPathToField column path to field for updated schema
+   * @param icebergCompatV2 Iceberg compatV2 is enabled
+   * @throws IllegalArgumentException in case of validation failure
+   */
+  static void validateColumnIds(
+      Map<String, StructField> currentColumnPathToField,
```

Comment:
do we want to consider a case-insensitive key here to avoid mismatches?

Reply:
We should at a later point yes. If I'm not mistaken in Delta we're case insensitive by default